bundle-workflow/src/test.py

```diff
@@ -6,49 +6,48 @@
 # this file be licensed under the Apache-2.0 license or a
 # compatible open source license.
 
-import argparse
 import os
 
-from git.git_repository import GitRepository
 from manifests.bundle_manifest import BundleManifest
 from paths.script_finder import ScriptFinder
 from system.temporary_directory import TemporaryDirectory
+from test_workflow.bwc_test_suite import BwcTestSuite
 from test_workflow.integ_test_suite import IntegTestSuite
 from test_workflow.local_test_cluster import LocalTestCluster
+from test_workflow.test_args import TestArgs
+from test_workflow.test_component import TestComponent
 
-parser = argparse.ArgumentParser(description="Test an OpenSearch Bundle")
-parser.add_argument("manifest", type=argparse.FileType("r"), help="Manifest file.")
-parser.add_argument(
-    "--keep",
-    dest="keep",
-    action="store_true",
-    help="Do not delete the working temporary directory.",
-)
-args = parser.parse_args()
-
+args = TestArgs()
 manifest = BundleManifest.from_file(args.manifest)
 script_finder = ScriptFinder()
 
-with TemporaryDirectory(keep=args.keep) as work_dir:
-    os.chdir(work_dir)
 
-    # Spin up a test cluster with security
-    # TODO: Refactor this into a TestRun class so we can run with-security and without-security without cut-and-pasting a bunch of code
-    cluster = LocalTestCluster(work_dir, manifest, True)
-    try:
+def integ_test_suite():
+    with TemporaryDirectory(keep=args.keep) as work_dir:
+        os.chdir(work_dir)
+
+        # Spin up a test cluster
+        cluster = LocalTestCluster(manifest)
         cluster.create()
 
         # For each component, check out the git repo and run `integtest.sh`
-        for component in manifest.components:
-            print(component.name)
-            repo = GitRepository(
-                component.repository,
-                component.commit_id,
-                os.path.join(work_dir, component.name),
-            )
-            test_suite = IntegTestSuite(component.name, repo, script_finder)
-            test_suite.execute(cluster, True)
-    finally:
-        cluster.destroy()
-
-    # TODO: Store test results, send notification.
+        try:
+            for component in manifest.components:
+                print(component.name)
+                test_component = TestComponent(component.repository, component.commit_id)
+                repo = test_component.checkout(os.path.join(work_dir, component.name))
+                test_suite = IntegTestSuite(component.name, repo)
+                test_suite.execute(cluster)
+        finally:
+            cluster.destroy()
+
+        # TODO: Store test results, send notification.
+
+
+def bwc_test_suite():
```

Comment:
Yeah agree that each job should have its own entry point. I'll go create `run_bwc_test.py` and `bwc_test.sh`
Yeah the idea this PR puts in is that, given a manifest and few options its takes care of all running all the suite of tests for all components. 

But thinking about it I like the idea of abstracting running the tests vs orchestrating the tests. 

The advantages of approach #338 proposes:
1. Tests for each component is abstracted into `bwc_test_suite.py` i.e each test run is abstracted and could have independent test recorder. 
2. Helps the runner to consolidate all the results into and if needed it could add retries later.

I'll merge this in and create a quick follow up to structure the code as we discussed here. 
What do you think?

Reply:
sounds good to me @saratvemulapalli